lms/djangoapps/courseware/module_render.py

```diff
@@ -1107,9 +1108,24 @@ def get_module_by_usage_id(request, course_id, usage_id, disable_staff_debug_inf
         tracking_context['module']['original_usage_key'] = str(descriptor_orig_usage_key)
         tracking_context['module']['original_usage_version'] = str(descriptor_orig_version)
 
-    unused_masquerade, user = setup_masquerade(request, course_id, has_access(user, 'staff', descriptor, course_id))
+    return descriptor, tracking_context
+
+
+def get_module_by_usage_id(request, course_id, usage_id, disable_staff_debug_info=False, course=None,
+                           will_recheck_access=False):
+    """
+    Gets a module instance based on its `usage_id` in a course, for a given request/user
+
+    Returns (instance, tracking_context)
+    """
+    course_key = CourseKey.from_string(course_id)
+    usage_key = _get_usage_key_for_course(course_key, usage_id)
+    descriptor, tracking_context = _get_descriptor_by_usage_key(usage_key)
+
+    user = request.user
```

Comment:
Why this line? Could just pass request.user into the has_access call and then you can drop it?

Reply:
Yeah it gets redefined immediately anyway. This was just old lines that became more obviously silly due to this refactor. Can fix by just passing request.user to the masquerade code.